### PR TITLE
WT-9447 Fix errors around drop and tiered storage from schema_abort.

### PR DESCRIPTION
**Pull request by sueloverso**
Mostly handle ENOENT in a few places.

### CODE REVIEW

**Comment by sueloverso** [comment]
This is ready for review. I have confirmed that the error I'm seeing is WT-9446 (i.e. a tiered->obj_config field is NULL).

**Review by AlexeyAnisimovAU — APPROVED** [review]
LGTM!

File: ext/storage_sources/dir_store/dir_store.c [hunk 1]
```diff
@@ -633,7 +633,12 @@ dir_store_file_copy(DIR_STORE *dir_store, WT_SESSION *session, const char *src_p
     }
     if ((ret = wt_fs->fs_open_file(wt_fs, session, src_path, type, WT_FS_OPEN_READONLY, &src)) !=
       0) {
-        ret = dir_store_err(dir_store, session, ret, "%s: cannot open for read", src_path);
+        /*
+         * It is normal and possible that the source file was dropped. Don't print out an error
+         * message in that case, but still return the ENOENT error value.
+         */
+        if (ret != ENOENT)
+            ret = dir_store_err(dir_store, session, ret, "%s: cannot open for read", src_path);
```
Comment (ddanderson) [inline ext/storage_sources/dir_store/dir_store.c:641]: `dir_store_file_copy` is called from two different places.  It looks like `dir_store_open` can validly return ENOENT, and we shouldn't complain.  However I _think_ if we ever get an error from via the call from `dir_store_flush`, we want a complaint.  Is that right?  I'm worried about having a weird ENOENT during a flush_tier call and not knowing where it came from.

**Review by ddanderson — APPROVED** [review]
LGTM but with one question in the dir_store module.

**Review by ddanderson — APPROVED** [review]
These latest changes LGTM

File: ext/storage_sources/dir_store/dir_store.c [hunk 2]
```diff
@@ -633,7 +633,12 @@ dir_store_file_copy(DIR_STORE *dir_store, WT_SESSION *session, const char *src_p
     }
     if ((ret = wt_fs->fs_open_file(wt_fs, session, src_path, type, WT_FS_OPEN_READONLY, &src)) !=
       0) {
-        ret = dir_store_err(dir_store, session, ret, "%s: cannot open for read", src_path);
+        /*
+         * It is normal and possible that the source file was dropped. Don't print out an error
+         * message in that case, but still return the ENOENT error value.
+         */
+        if ((ret != 0 && ret != ENOENT) || (ret == ENOENT && !enoent_okay))
+            ret = dir_store_err(dir_store, session, ret, "%s: cannot open for read", src_path);
```
Comment (sueloverso) [inline ext/storage_sources/dir_store/dir_store.c:641]: @ddanderson please review this.

**Pull request merged**